lib/logstash/outputs/elasticsearch/template_manager.rb

```diff
@@ -1,8 +1,18 @@
 module LogStash; module Outputs; class ElasticSearch
   class TemplateManager
+    LEGACY_TEMPLATE_ENDPOINT = '_template'.freeze
+    INDEX_TEMPLATE_ENDPOINT = '_index_template'.freeze
+
     # To be mixed into the elasticsearch plugin base
     def self.install_template(plugin)
       return unless plugin.manage_template
+
+      if plugin.maximum_seen_major_version < 8 && plugin.template_api == 'auto'
+        plugin.logger.warn("Elasticsearch Output configured with `template_api => auto` uses legacy template API to manage index template for Elasticsearch 7 or below. " +
+                             "Legacy template API will be removed in Elasticsearch 9. It is recommended to set `template_api => legacy` before any upgrade " +
+                             "and please consider to migrate to composable index templates.")
```

Comment:
I've reworked this with a couple things in mind.
1. the logger name already indicates that this is coming from the Elasticsearch Output, so I avoided starting with that.
2. I brought forward _why_ we resolved to `legacy`, and indicated what it will resolve to when Logstash connects to ES 8
3. I defined the two courses of action, either of which will silence this message
   - ``setting `template_api => legacy` to continue providing legacy-style templates`` OR
   - ``migrating your template to the composable style and setting `template_api => composable`.``
4. I indicated that the legacy template AP will be removed in _Elasticsearch 9_.

```suggestion
        plugin.logger.warn("`template_api => auto` resolved to `legacy` since we are connected to " + "Elasticsearch #{plugin.maximum_seen_major_version}, " +
                           "but will resolve to `composable` the first time it connects to Elasticsearch 8+. " +
                           "We recommend either setting `template_api => legacy` to continue providing legacy-style templates, " +
                           "or migrating your template to the composable style and setting `template_api => composable`. " +
                           "The legacy template API is slated for removal in Elasticsearch 9.")
```